### PR TITLE
Add rpm builder for VMware Photon OS

### PR DESCRIPTION
**Pull request by frapposelli**
Signed-off-by: Fabio Rapposelli fabio@vmware.com

### CODE REVIEW

File: contrib/builder/rpm/amd64/generate.sh [hunk 1]
```diff
@@ -58,6 +61,10 @@ for version in "${versions[@]}"; do
 			# get rpm-build and curl packages and dependencies
 			echo 'RUN zypper --non-interactive install ca-certificates* curl gzip rpm-build' >> "$version/Dockerfile"
 			;;
+		photon:*)
+			echo "RUN ${installer} install -y wget git gzip rpm-build" >> "$version/Dockerfile"
+			echo "RUN wget -qO- https://gist.githubusercontent.com/toliaqat/686a7f05b0dcd006296a/raw/25592008aa215f6d8fe031536522315ac063f978/build-essential.txt | xargs ${installer} install -y" >> "$version/Dockerfile"
```
Comment (jfrazelle) [inline contrib/builder/rpm/amd64/generate.sh]: we can't rely on a script hosted by a third party

Comment (tianon) [inline contrib/builder/rpm/amd64/generate.sh]: Yeah :disappointed: 

There's not a metapackage or group or something to just pull all these in?  That's a reallly long list. :cry:

File: contrib/builder/rpm/amd64/photon-1.0RC/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#
+# THIS FILE IS AUTOGENERATED; SEE "contrib/builder/rpm/amd64/generate.sh"!
+#
+
+FROM photon:1.0RC
```
Comment (tianon) [inline contrib/builder/rpm/amd64/photon-1.0RC/Dockerfile]: How long will `1.0RC` be supported?  What impact will this have on the release script (https://github.com/docker/docker/blob/master/hack/make/release-rpm) and artifacts after 1.0 becomes GA, etc?

File: contrib/builder/rpm/amd64/generate.sh (hunk 1 above)
Comment (frapposelli) [inline contrib/builder/rpm/amd64/generate.sh]: yeah, I know :disappointed:, a metapackage is coming soon, but I thought I would get the ball rolling in the meantime.

File: contrib/builder/rpm/amd64/photon-1.0RC/Dockerfile (hunk 2 above)
Comment (frapposelli) [inline contrib/builder/rpm/amd64/photon-1.0RC/Dockerfile]: `1.0RC` is going to be supported until GA comes out (sometimes later this year), once GA is out we can decide to either keep RC artifacts around for another 6 months or just drop them in favor of GA.

**Comment by jfrazelle** [comment]
is there a reason why we should support this now or can we wait for GA

**Comment by thaJeztah** [comment]
Agreed, I'd prefer to wait for a stable/GA release before merging this.

**Comment by jfrazelle** [comment]
I'm going to close, thanks so much, hope its soon!

**Pull request closed without merging**